include/albatross/src/indexing/group_by.hpp

```diff
@@ -174,12 +190,40 @@ template <typename KeyType, typename ValueType> class GroupedBase {
   std::map<KeyType, ValueType> map_;
 };
 
-template <typename KeyType, typename ValueType>
+template <typename KeyType, typename ValueType, typename Enable>
 class Grouped : public GroupedBase<KeyType, ValueType> {
   using Base = GroupedBase<KeyType, ValueType>;
   using Base::Base;
 };
 
+/*
+ * Some specialized functionality for arithmetic value types.
+ */
+template <typename KeyType, typename ValueType>
+class Grouped<KeyType, ValueType,
+              std::enable_if_t<std::is_arithmetic<ValueType>::value>>
+    : public GroupedBase<KeyType, ValueType> {
+  using Base = GroupedBase<KeyType, ValueType>;
+  using Base::Base;
+
+public:
+  auto sum() const {
+    double output = 0.;
```

Comment:
Wouldn't you want this type to be `ValueType`? In the case of large-ish integer values you'd end up losing precision by internally using a `double` to contain the sum.

Reply:
yeah, good call.